server/core/runtime/state_rm_step_runner.go

```diff
@@ -0,0 +1,38 @@
+package runtime
+
+import (
+	"os"
+	"path/filepath"
+
+	version "github.com/hashicorp/go-version"
+	"github.com/runatlantis/atlantis/server/events/command"
+)
+
+type StateRmStepRunner struct {
+	TerraformExecutor TerraformExec
+	DefaultTFVersion  *version.Version
+}
+
+func (p *StateRmStepRunner) Run(ctx command.ProjectContext, extraArgs []string, path string, envs map[string]string) (string, error) {
+	tfVersion := p.DefaultTFVersion
+	if ctx.TerraformVersion != nil {
+		tfVersion = ctx.TerraformVersion
+	}
+
+	stateRmCmd := []string{"state", "rm"}
+	stateRmCmd = append(stateRmCmd, extraArgs...)
+	stateRmCmd = append(stateRmCmd, ctx.EscapedCommentArgs...)
+	out, err := p.TerraformExecutor.RunCommandWithVersion(ctx, filepath.Clean(path), stateRmCmd, envs, tfVersion, ctx.Workspace)
+
+	// If the state rm was successful and a plan file exists, delete the plan.
```

Comment:
We should document this in the .md docs

Reply:
docs is written in https://deploy-preview-2880--runatlantis.netlify.app/docs/using-atlantis.html#atlantis-state-rm, like `This command discards the terraform plan result. After run state rm and before an apply, another atlantis plan must be run again.`

In addition, I added vcs comment about discarding like [notify import/state rm discard plan file](https://github.com/runatlantis/atlantis/pull/2880/commits/a6ad798a86d449cd2c491b6c609116e6d14d5a4b)